File: db2/src/test/java/com/dbx/agent/db2/Db2AgentTest.java
package com.dbx.agent.db2;

import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.ConnectParams;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Db2AgentTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new Db2Agent();
    }

    @Override
    protected String resultSetSql() {
        return "CALL ADMIN_CMD('list applications')";
    }

    @Test
    void buildsJdbcUrlWithDb2PropertySuffix() {
        ConnectParams params = new ConnectParams();
        params.setHost("db.example.com");
        params.setPort(50000);
        params.setDatabase("SAMPLE");
        params.setUrl_params("sslConnection=true");

        assertEquals("jdbc:db2://db.example.com:50000/SAMPLE:sslConnection=true;", Db2Agent.buildUrl(params));
    }
}

File: db2/src/test/kotlin/com/dbx/agent/db2/Db2AgentTest.kt
package com.dbx.agent.db2

import com.dbx.agent.test.JdbcAgentFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class Db2AgentTest {
    @Test
    fun `executes non select statements that return result sets`() {
        val agent = Db2Agent()
        setPrivateConnection(agent, JdbcAgentFake.connection())

        val result = agent.executeQuery("CALL ADMIN_CMD('list applications')", null)

        assertEquals(listOf("VALUE"), result.columns)
        assertEquals(listOf(listOf("row-value")), result.rows)
        assertEquals(listOf("execute"), JdbcAgentFake.calls)
    }
}
